diff --git a/node_modules/@whiskeysockets/baileys/lib/Socket/socket.js b/node_modules/@whiskeysockets/baileys/lib/Socket/socket.js
index 6357f9e..f191040 100644
--- a/node_modules/@whiskeysockets/baileys/lib/Socket/socket.js
+++ b/node_modules/@whiskeysockets/baileys/lib/Socket/socket.js
@@ -601,7 +601,7 @@ export const makeSocket = (config) => {
             name: '~'
         };
         ev.emit('creds.update', authState.creds);
-        await sendNode({
+        const result = await query({
             tag: 'iq',
             attrs: {
                 to: S_WHATSAPP_NET,
@@ -646,7 +646,10 @@ export const makeSocket = (config) => {
                     ]
                 }
             ]
-        });
+        }, 8000);
+        if (result?.attrs?.type === 'error') {
+            throw new Boom('Permintaan kode pairing ditolak. Silakan coba lagi nanti.', { statusCode: 400 });
+        }
         return authState.creds.pairingCode;
     };
     async function generatePairingKey() {
